plugins/modules/ipatrust.py

```diff
@@ -0,0 +1,282 @@
+            admin=dict(type="str", default=None, required=False),
+            passwordd=dict(type="str", default=None,
+                           required=False, no_log=True),
+            server=dict(type="str", default=None, required=False),
+            trust_secret=dict(type="str", default=None,
+                              required=False, no_log=True),
+            base_id=dict(type="int", default=None, required=False),
+            range_size=dict(type="int", default=200000, required=False),
+            range_type=dict(type="str", default="ipa-ad-trust", required=False, choices=[
+                            "ipa-ad-trust-posix", "ipa-ad-trust"]),
+            two_way=dict(type="bool", default=False, required=False),
+            external=dict(type="bool", default=False, required=False),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # general
+    ipaadmin_principal = module_params_get(
+        ansible_module, "ipaadmin_principal")
+    ipaadmin_password = module_params_get(ansible_module, "ipaadmin_password")
+    realm = module_params_get(ansible_module, "realm")
+
+    # state
+    state = module_params_get(ansible_module, "state")
+
+    # trust
+    trust_type = module_params_get(ansible_module, "trust_type")
+    realm_admin = module_params_get(ansible_module, "admin")
+    realm_passwd = module_params_get(ansible_module, "password")
+    realm_server = module_params_get(ansible_module, "server")
+    trust_secret = module_params_get(ansible_module, "trust_secret")
+    base_id = module_params_get(ansible_module, "base_id")
+    range_size = module_params_get(ansible_module, "range_size")
+    range_type = module_params_get(ansible_module, "range_type")
+    bidirectional = module_params_get(ansible_module, "two_way")
+    external = module_params_get(ansible_module, "external")
+
+    changed = False
+    exit_args = {}
+    ccache_dir = None
+    ccache_name = None
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(
+                ipaadmin_principal, ipaadmin_password)
+        api_connect()
+        res_find = find_trust(ansible_module, realm)
+
+        if state == "absent" and res_find is not None:
+            del_trust(ansible_module, realm)
+            changed = True
+
+        if state == "present" and res_find is None:
+            exclusive = ["realm_admin", "realm_passwd"]
+            if vars()['trust_secret'] is not None:
+                for x in exclusive:
+                    if vars()[x] is not None:
+                        ansible_module.fail_json(
+                            msg="Argument '%s' is mutually exclusive with trust_secret" % (x))
+            elif vars()['realm_admin'] is None:
+                ansible_module.fail_json(
+                    msg="State present requires trust-secret or admin to be set")
+            elif vars()['realm_passwd'] is None:
```

Comment:
Check module `ipahost` and `ipavault` on how they handle mutually exclusive and "required one of many" arguments.

Reply:
done